IMG_bmp.c: GetIconSurface returns false triggering -Wbool-conversion

In `src/IMG_bmp.c`, `GetIconSurface` is declared to return `SDL_Surface *`, but on one error path it returns `false` instead of a null pointer. This triggers the -Wbool-conversion warning.

```c
static SDL_Surface *GetIconSurface(SDL_IOStream *src, Sint64 offset, int type, int hot_x, int hot_y)
{
    SDL_Surface *surface = NULL;
    Uint32 biSize;

    if (SDL_SeekIO(src, offset, SDL_IO_SEEK_SET) < 0) {
        return false;
    }

    ...
}
```
Link to offending line (also to quickly see context):
https://github.com/libsdl-org/SDL_image/blob/22dd4dcc5c80081d5ecda241cf1fa225bdeb85ad/src/IMG_bmp.c#L537

With Clang, this produces:
```
SDL_image/src/IMG_bmp.c:537:16: warning: initialization of pointer of type 'SDL_Surface *' (aka 'struct SDL_Surface *') to null from a constant boolean expression [-Wbool-conversion]
  537 |         return false;
      |                ^~~~~
1 warning generated.
```

For context (if relevant), observed with apple clang.

--- a/src/IMG_bmp.c
+++ b/src/IMG_bmp.c
@@ -534,7 +534,7 @@ static SDL_Surface *GetIconSurface(SDL_IOStream *src, Sint64 offset, int type, i
     Uint32 biSize;
 
     if (SDL_SeekIO(src, offset, SDL_IO_SEEK_SET) < 0) {
-        return false;
+        return NULL;
     }
 
     if (!SDL_ReadU32LE(src, &biSize) ||
@@ -948,4 +948,3 @@ bool IMG_SaveICO(SDL_Surface *surface, const char *file)
 }
 
 #endif // SAVE_BMP
-

